client/api/volume/v1beta3/api.proto

```diff
@@ -10,27 +10,52 @@ service Volume {
+    // UnmountVolume gracefully unmounts a volume
+    rpc UnmountVolume(UnmountVolumeRequest) returns (UnmountVolumeResponse) {}
+    // IsVolumeFormatted checks if a volume is formatted.
     rpc IsVolumeFormatted(IsVolumeFormattedRequest) returns (IsVolumeFormattedResponse) {}
-    // FormatVolume formats a volume with the provided file system
+    // FormatVolume formats a volume with NTFS.
     rpc FormatVolume(FormatVolumeRequest) returns (FormatVolumeResponse) {}
     // ResizeVolume performs resizing of the partition and file system for a block based volume
     rpc ResizeVolume(ResizeVolumeRequest) returns (ResizeVolumeResponse) {}
+    // GetVolumeStats gathers DiskSize, VolumeSize and VolumeUsedSize for a volume
+    rpc GetVolumeStats(GetVolumeStatsRequest) returns (GetVolumeStatsResponse) {}
+    // GetDiskNumberFromVolumeID gets the disk number of the disk where the volume is located
+    rpc GetDiskNumberFromVolumeID(GetDiskNumberFromVolumeIDRequest) returns (GetDiskNumberFromVolumeIDResponse ) {}
+    // GetVolumeIDFromTargetPath gets the volume id for a given mount
+    rpc GetVolumeIDFromTargetPath(GetVolumeIDFromTargetPathRequest) returns (GetVolumeIDFromTargetPathResponse) {}
+    // WriteVolumeCache write volume cache to disk
+    rpc WriteVolumeCache(WriteVolumeCacheRequest) returns (WriteVolumeCacheResponse) {}
+
+    // Deprecated apis
+
+    // DismountVolume dismounts a volume
+    rpc DismountVolume(DismountVolumeRequest) returns (DismountVolumeResponse) {
+        // Use UnmountVolume instead
+        option deprecated = true;
+    }
     // VolumeStats gathers DiskSize, VolumeSize and VolumeUsedSize for a volume
-    rpc VolumeStats(VolumeStatsRequest) returns (VolumeStatsResponse) {}
+    rpc VolumeStats(VolumeStatsRequest) returns (VolumeStatsResponse) {
+        // Renamed to GetVolumeStats
+        option deprecated = true;
+    }
     // GetVolumeDiskNumber gets the disk number of the disk where the volume is located
-    rpc GetVolumeDiskNumber(VolumeDiskNumberRequest) returns (VolumeDiskNumberResponse) {}
+    rpc GetVolumeDiskNumber(VolumeDiskNumberRequest) returns (VolumeDiskNumberResponse) {
+        // Renamed to GetDiskNumberFromVolumeID
+        option deprecated = true;
+    }
     // GetVolumeIDFromMount gets the volume id for a given mount
-    rpc GetVolumeIDFromMount(VolumeIDFromMountRequest) returns (VolumeIDFromMountResponse) {}
-    // WriteVolumeCache write volume cache to disk
-    rpc WriteVolumeCache(WriteVolumeCacheRequest) returns (WriteVolumeCacheResponse) {}
+    rpc GetVolumeIDFromMount(VolumeIDFromMountRequest) returns (VolumeIDFromMountResponse) {
+        // Renamed to GetVolumeIDFromTargetPath
+        option deprecated = true;
+    }
 }
 
 message ListVolumesOnDiskRequest {
-    // Disk device ID of the disk to query for volumes
-    string disk_id = 1;
+    // Disk device number of the disk to query for volumes
+    int64 disk_number = 1;
```

Comment:
i think int is enough

Reply:
I agree that `int32` is enough however I'd like to keep everything as `int64` because all of the other int like fields in this file are `int64`

Reply:
I take that back, I'm going to change it to `uint32` as that's the field type returned by `GetDisk`